crosslink/README.md

```diff
@@ -0,0 +1,128 @@
+root flag is not provided, crosslink will attempt to find the root directory of
+a git repository using a tool available in the go-build-tools repo. The root
+flag is available to all crosslink subcommands.
+
+**Note: If no --root flag is provided than crosslink attempts to identify a git
+repository in the current or a parent directory. If no git repository exists
+crosslink will panic.**
+
+    crosslink --root=/users/foo/multimodule-go-repo
+
+### prune / –-prune
+
+`CAUTION: DESTRUCTIVE`
+
+The prune command or flag will run the prune action on the current dependency.
+Pruning will remove any dependencies that are not in the current
+intra-repository dependency graph. Pruning will only remove go modules that
+fall under the same module path as the root module. For example,
+If the root module is named `example.com/foo` and there exists a replace
+statement of `example.com/foo/bar => ./bar` that is not a direct or transitive
+dependency of the current go.mod file, it will be pruned.
+
+**Crosslink will not remove replace statements for modules that do not
+fall under the root module path even if they are not in the current
+dependency graph.**
+
+Pruning can be executed independently with no replace statements being inserted.
+
+    crosslink prune
+
+Pruning can also be executed in addition to a standard crosslink replace operation.
+
+    crosslink --root=/users/foo/multimodule-go-repo --prune
+
+### –-overwrite
+
+`CAUTION: DESTRUCTIVE`
+
+Overwrite gives crosslink permission to update existing replacement paths
+in addition to adding new ones. If crosslink identifies a direct or
+transitive dependency in the intra-repository graph then it will insert
+or update the corresponding replace statement for that requirement.
+
+    crosslink --root=/users/foo/multimodule-go-repo --overwrite
+
+### –-exclude
+
+Exclude is a set of go modules that crosslink will ignore when replacing or pruning.
+It is expected that a list of comma separated values will be provided in one or
+multiple calls to exclude. Excluded module names should be the old value in the
+replacement path. For example, passing `example.com/test` would exclude this replace
+statement from any operation `replace example.com/test => ./test`.
+
+A single call to exclude
+    crosslink --overwrite --exclude=github.com/foo/bar/modA,github.com/foo/bar/modB
+
+Multiple calls to exclude can also be made
+
+    crosslink –prune --exclude=example.com/foo/bar/modA,example.com/foo/bar/modB\
+    --exclude=example.com/foo/bar/modC \
+    --exclude=example.com/foo/bar/modJ,example.com/modZ
+
+### –-verbose
+
+Verbose enables crosslink to log all replace (destructive and non-destructive) and
+pruning operations to the terminal. By default this is disabled but enabled
+automatically when the –overwrite flag is provided. Verbosity can be enabled or
+disabled at any time.
+For non destructive operations
+
+    crosslink --root=/users/foo/multimodule-go-repo -v
```

Comment:
This shows a short option. Do all of these options have equivalent short options, and if say can you update the section headers to include the short options as well as the long ones?

Reply:
They do not. Currently only `verbose` and `overwrite` have short options. I will update the section headers for those two. I am unsure on what the best practice here is, is it sensible for all commands to have short options?